plugins/modules/openshift_build.py

```diff
@@ -0,0 +1,268 @@
+    namespace: "default"
+    build_name: ruby-build-1
+    state: cancelled
+
+# Cancel the named build and create a new one with the same parameters
+- name: Cancel build from default namespace and create a new one
+  community.okd.openshift_cancel_build:
+    namespace: "default"
+    build_name: ruby-build-1
+    state: restarted
+
+# Cancel all builds created from 'ruby-build' build configuration that are in 'new' state
+- name: Cancel build from default namespace and create a new one
+  community.okd.openshift_cancel_build:
+    namespace: "default"
+    build_config_name: ruby-build
+    build_phase:
+      - New
+    state: cancelled
+'''
+
+RETURN = r'''
+builds:
+  description:
+  - The builds that were started/cancelled.
+  returned: success
+  type: complex
+  contains:
+    api_version:
+      description: The versioned schema of this representation of an object.
+      returned: success
+      type: str
+    kind:
+      description: Represents the REST resource this object represents.
+      returned: success
+      type: str
+    metadata:
+      description: Standard object metadata. Includes name, namespace, annotations, labels, etc.
+      returned: success
+      type: dict
+    spec:
+      description: Specific attributes of the build.
+      returned: success
+      type: dict
+    status:
+      description: Current status details for the object.
+      returned: success
+      type: dict
+'''
+# ENDREMOVE (downstream)
+
+import copy
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.kubernetes.core.plugins.module_utils.args_common import AUTH_ARG_SPEC
+from ansible.module_utils._text import to_native
+
+
+def argument_spec():
+    args = copy.deepcopy(AUTH_ARG_SPEC)
+
+    args_options = dict(
+        name=dict(type='str', required=True),
+        value=dict(type='str', required=True)
+    )
+
+    args.update(
+        dict(
+            state=dict(type='str', choices=['started', 'cancelled', 'restarted'], default="started"),
+            build_args=dict(type='list', elements='dict', options=args_options),
+            commit=dict(type='str'),
+            env_vars=dict(type='list', elements='dict', options=args_options),
+            build_name=dict(type='str'),
+            build_config_name=dict(type='str'),
+            namespace=dict(type='str', required=True),
+            incremental=dict(type='bool'),
+            no_cache=dict(type='bool'),
+            wait=dict(type='bool', default=False),
+            wait_sleep=dict(type='int', default=5),
+            wait_timeout=dict(type='int', default=120),
+            build_phases=dict(type='list', elements='str', default=[]),
+        )
+    )
+    return args
+
+
+def main():
+    mutually_exclusive = [
+        ('build_name', 'build_config_name'),
+    ]
+    module = AnsibleModule(
+        argument_spec=argument_spec(),
+        mutually_exclusive=mutually_exclusive,
+        required_one_of=[
```

Comment:
I think it is not because none of these parameters are mandatory, if we remove required_one_of, this would mean that user can call the module with neither the build name or the build config name

Reply:
Right!